lisa/env.py

```diff
@@ -210,6 +212,46 @@ def from_one_conf(cls, path):
             plat_info = None
         return cls(target_conf=target_conf, plat_info=plat_info)
 
+    @classmethod
+    def from_cli(cls):
+        """
+        Create a TestEnv from command line arguments
+        """
+        description = textwrap.dedent(
+            """
+            Extra arguments differ depending on the value passed to 'kind'.
+            Try e.g. \"--kind android\" to see the arguments for android targets.
+            """
+        )
+        parser = argparse.ArgumentParser(description=description,
+                                         formatter_class=argparse.RawDescriptionHelpFormatter)
+        parser.add_argument("--kind", "-k",
+                            choices=["android", "linux", "host"],
+                            required=True,
+                            help="The kind of target to create")
+
+        # subparsers cannot let us specify --kind=android, at best we could
+        # have --android which is lousy.
+        # Use parse_known_args() instead to look at --kind, and add relevent
+        # arguments
+        args = parser.parse_known_args()[0]
```

Comment:
The comment on subparsers is right. The main issue with parsing twice like this is that --help/-h will be caught by parser.parse_known_args which will not provide a helpful help message and then exit. The solution is to just add all parameters to the parser, and only use the one that are relevant for a given kind (and specify that in the help strings of options).